apps/heft/src/plugins/JestPlugin/JestPlugin.ts

```diff
@@ -145,7 +142,125 @@ export class JestPlugin implements IHeftPlugin {
     clean.properties.pathsToDelete.add(cacheFolder);
   }
 
+  private async _getJestReporters(
+    heftSession: HeftSession,
+    heftConfiguration: HeftConfiguration,
+    jestLogger: ScopedLogger
+  ): Promise<JestReporterConfig[]> {
+    const config: Config.GlobalConfig = await JsonFile.loadAsync(this._getJestConfigPath(heftConfiguration));
+    let reporters: JestReporterConfig[];
+    let isUsingHeftReporter: boolean = false;
+    let parsedConfig: boolean = false;
+
+    if (Array.isArray(config.reporters)) {
+      reporters = config.reporters;
+
+      // Harvest all the array indices that need to modified before altering the array
+      const heftReporterIndices: number[] = this._findIndexes(config.reporters, 'default');
+      const jestDefaultReporterIndexes: number[] = this._findIndexes(config.reporters, '__jest_default');
```

Comment:
@iclanton said (in a separate comment):
> I just built this locally and it looks like the default Jest reporter is writing to STDERR, causing the test project to succeed with warnings. @octogonz - thoughts on how we should handle this?

The `__jest_default` constant does not seem to be part of Jest.  I'm thinking it's an invented symbol for referencing the real Jest default reporter (instead of Heft's default reporter). This is used `heft-jest-reporters-test` probably to avoid the hassle of making a custom reporter:

```js
{
  "preset": "./node_modules/@rushstack/heft/includes/jest-shared.config.json",
  "reporters": ["default", "__jest_default"]
}
```

But as you pointed out, Jest's default reporter is not very compatible with Rush because it writes to STDERR.

@ifeanyiecheruo do you actually need `__jest_default` for any real purpose?  If not, maybe we should delete this `__jest_default` feature, and `heft-jest-reporters-test` can define a trivial reporter for testing purposes.

Reply:
You guessed right. __jest_default is a back door to get the internal jest reporter which itself is not directly exported.
If it is fundamentally in conflict with Rush then I'll yank it